### PR TITLE
feat(core): it now is possible to ignore_access during elgg_get_entities

### LINKED ISSUE #10401: ignore access as part of elgg_get_entities
we added a helper flag to ignore access in get_entity... should we do the same in elgg_get_entities? There are a lot of usecases where we ignore access to fetch entities...

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #10401

### CODE REVIEW

File: engine/classes/Elgg/Database/EntityTable.php [hunk 1]
```diff
@@ -552,6 +556,10 @@ public function getEntities(array $options = array()) {
 
 			return new \ElggBatch([$this, 'getEntities'], $options, null, $batch_size, $batch_inc_offset);
 		}
+		
+		if ($options['ignore_access']) {
+			$old_ia = elgg_set_ignore_access(true);
```
Comment (juho-jaakkola) [inline engine/classes/Elgg/Database/EntityTable.php]:
```php
$this->session->setIgnoreAccess(true);
```

File: engine/classes/Elgg/Database/EntityTable.php [hunk 2]
```diff
@@ -719,6 +727,10 @@ public function getEntities(array $options = array()) {
 				_elgg_services()->entityPreloader->preload($results, $props_to_preload);
 			}
 		}
+		
+		if ($options['ignore_access']) {
+			elgg_set_ignore_access($old_ia);
```
Comment (juho-jaakkola) [inline engine/classes/Elgg/Database/EntityTable.php]:
```php
$this->session->setIgnoreAccess($old_ia);
```

**Review by juho-jaakkola — CHANGES_REQUESTED** [review]

**Comment by hypeJunction** [comment]
I don't think this will work either way unless all metadata is loaded with the entity. The sql query will only be built with ignored access for the entity table row. Speaking from experience

**Comment by jdalsem** [comment]
Closing this. It works (for some usecases) but for others you still need the 'old' pattern of ignoring access. To prevent confusion on how it works, we better stick with one approach.

**Pull request closed without merging**